apps/server/src/shared/infra/identity-management/keycloak-identity-management.spec.ts

```diff
@@ -0,0 +1,255 @@
+			expect(deleteAccountMock).not.toHaveBeenCalled();
+		});
+	});
+
+	describe('findAccountById', () => {
+		it('should find an existing account', async () => {
+			const accountId = 'user-1-id';
+			const accountUserName = 'user-1';
+			const accountEmail = 'user@mail';
+			const accountFirstName = 'user';
+			const accountLastName = '1';
+
+			jest.spyOn(kcAdminClient.users, 'findOne').mockResolvedValue({
+				id: accountId,
+				username: accountUserName,
+				email: accountEmail,
+				firstName: accountFirstName,
+				lastName: accountLastName,
+			});
+
+			const ret = await idm.findAccountById(accountId);
+
+			expect(ret).not.toBeNull();
+			expect(ret).toEqual(
+				expect.objectContaining({
+					id: accountId,
+					userName: accountUserName,
+					email: accountEmail,
+					firstName: accountFirstName,
+					lastName: accountLastName,
+				})
+			);
+		});
+
+		it('should reject if account does not exist', async () => {
+			jest.spyOn(kcAdminClient.users, 'findOne').mockRejectedValue('error');
+			await expect(idm.findAccountById('accountId')).rejects.toBeTruthy();
+
+			jest.spyOn(kcAdminClient.users, 'findOne').mockResolvedValue(undefined);
+			await expect(idm.findAccountById('accountId')).rejects.toBeTruthy();
+		});
+	});
+
+	describe('getAllAccounts', () => {
+		it('should find all existing accounts', async () => {
+			const accountId = 'user-1-id';
+			const accountUserName = 'user-1';
+			const accountEmail = 'user@mail';
+			const accountFirstName = 'user';
+			const accountLastName = '1';
+
+			jest.spyOn(kcAdminClient.users, 'find').mockResolvedValue([
+				{
+					id: accountId,
+					username: accountUserName,
+					email: accountEmail,
+					firstName: accountFirstName,
+					lastName: accountLastName,
+				},
+			]);
+
+			const ret = await idm.getAllAccounts();
+
+			expect(ret).not.toBeNull();
+			expect(ret).toHaveLength(1);
+			expect(ret).toContainEqual(
+				expect.objectContaining({
+					id: accountId,
+					userName: accountUserName,
+					email: accountEmail,
+					firstName: accountFirstName,
+					lastName: accountLastName,
+				})
+			);
+		});
+
+		it('should reject if loading all accounts failed', async () => {
+			jest.spyOn(kcAdminClient.users, 'find').mockRejectedValue('error');
+
+			await expect(idm.getAllAccounts()).rejects.toBeTruthy();
+		});
+	});
+
+	describe('updateAccount', () => {
+		it('should allow to update an existing account', async () => {
+			const accountId = 'user-1-id';
+			const updateUserMock = jest.spyOn(kcAdminClient.users, 'update').mockResolvedValue();
+			const testAccount = { firstName: 'test', email: 'test@test.test' };
+
+			const ret = await idm.updateAccount(accountId, testAccount);
+
+			expect(ret).not.toBeNull();
+			expect(ret).toBe(accountId);
+			expect(updateUserMock).toBeCalledWith(
+				expect.objectContaining({ id: accountId }),
+				expect.objectContaining({ firstName: testAccount.firstName, email: testAccount.email })
```

Comment:
Here using the spread operator is much shorter:
`expect.objectContaining({ ...testAccount })`

Reply:
These objects are different (internal Keycloak representation and IAccount transfer object). They just have a similar fields 😄